packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2851,6 +2842,43 @@ export class ContainerRuntime
 		}
 	}
 
+	private scheduleFlush() {
+		if (this.flushTaskExists) {
+			return;
+		}
+
+		this.flushTaskExists = true;
+		const flush = () => {
+			this.flushTaskExists = false;
+			this.flush();
+		};
+
+		switch (this.flushMode) {
+			case FlushMode.TurnBased:
+				Promise.resolve()
+					.then(() => flush())
+					.catch((error) => {
+						this.closeFn(error as GenericError);
+					});
+				break;
+			case FlushMode.Async:
```

Comment:
nit: can you add comment here explaining why Async uses setTimeout? Bascially, the description about using macro-task to achieve what the comment in the definition says:
"When in Async flush mode, the runtime will accumulate all operations across JS turns and send them as a single
 batch when all micro-tasks are complete."